exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/BaseIndexProxyTestable.java

```diff
@@ -197,7 +196,8 @@ void indexConstructorPersistsIndexTypeInfo() throws CloseFailuresException {
       Exception e = assertThrows(RuntimeException.class, () -> createOfOtherType(name, snapshot));
 
       // TODO: Change message after https://jira.bf.local/browse/ECR-3354
-      assertThat(e, hasMessage(containsString("Index type doesn't match specified")));
+      assertThat(e, hasMessage(containsStringIgnoringCase(
+              "Index type does not match specified one")));
```

Comment:
After ECR-3418 we have some changes in error messages. For the Entry case it is now _"Error while reading the index state. Possibly the index type does not match specified one"_, for the rest of indices it is _"Index type does not match specified one"_. @dmitry-timofeev is this solution acceptable?

Reply:
I think it is not the solution of the initial problem, as it does not show even the actual type of the index, does it?

Reply:
Well, at line 198 there is a //TODO with link to the task I created already for this problem - https://jira.bf.local/browse/ECR-3354

Reply:
Ah, sorry, I thought that this PR is actually aimed to fix ECR-3354 :D

Reply:
What we can see as an error output for tests on indices other than `Entry`:
```
"assertion failed: `(left == right)`
  left: `KeySet`,
 right: `List`: Index type does not match specified one"
```
So, in fact one can see the index type, but as part of a wrong assertion.

Reply:
I think yes, the alternative is too complicated to bother (add an assumption here that index type is not an entry to exclude it + an entry-only test).

Reply:
Do those left and right get into the message?

Reply:
Yes, it is the message we get from assertion in the `metadata::index_metadata()`:
```
        assert_eq!(
            metadata.index_type, index_type,
            "Index type does not match specified one"
        );
```

Reply:
and into the message of a Java Exception?

Reply:
Yes. `Index::new()` obviously panics on native assert -> we catch it and throw as a `RuntimeException`.

Reply:
So ECR-3354 is to make it clear which type is expected?

Reply:
I don't remember the details of the discussion that caused this ticket, but in general - yes.